addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/ExtensionAuthhelper.java

```diff
@@ -51,11 +53,36 @@ public class ExtensionAuthhelper extends ExtensionAdaptor {
 
     private static final Logger LOGGER = LogManager.getLogger(ExtensionAuthhelper.class);
 
+    private static final HeaderBasedSessionManagementMethodType HEADER_BASED_TYPE =
+            new HeaderBasedSessionManagementMethodType();
+
     public ExtensionAuthhelper() {
         super();
         this.setI18nPrefix("authhelper");
     }
 
+    @Override
+    public void optionsLoaded() {
+        ExtensionSessionManagement extSm =
+                Control.getSingleton()
+                        .getExtensionLoader()
+                        .getExtension(ExtensionSessionManagement.class);
+        if (extSm != null) {
+            extSm.getSessionManagementMethodTypes().add(HEADER_BASED_TYPE);
+        }
+    }
+
+    @Override
+    public void unload() {
+        ExtensionSessionManagement extSm =
+                Control.getSingleton()
+                        .getExtensionLoader()
+                        .getExtension(ExtensionSessionManagement.class);
+        if (extSm != null) {
+            extSm.getSessionManagementMethodTypes().remove(HEADER_BASED_TYPE);
```

Comment:
Extract a method that gets the add-on? Or even get the add-on and pass a boolean for add/remove?

Reply:
Could do .. but still need to check if its null each time. I'm not keen on booleans for add/remove type ops :/

Reply:
Well it could at least be simplified to:

```java
...
    if (getExtensionSessionManagement) {
        // add or remove
    }
...

private static boolean getExtensionSessionManagement() {
    ExtensionSessionManagement extSm =
            Control.getSingleton()
                    .getExtensionLoader()
                    .getExtension(ExtensionSessionManagement.class);
    if (extSm == null) {
       // log something
        return false;
    }
    return true;
}
```

Reply:
Changed 😁